sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -743,7 +743,11 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
         case a: AggregateExpression => a
       })
       if (aggregates.isEmpty) {
-        failAnalysis("The output of a correlated scalar subquery must be aggregated")
+        throw new AnalysisException(
```

Comment:
The current approach is to raise exception from `Query*Errors`. Could move this to `QueryCompilationErrors`, please.